pkg/tfcli/destroy.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tfcli
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	cliErrors "github.com/crossplane-contrib/terrajet/pkg/tfcli/errors"
+	"github.com/crossplane-contrib/terrajet/pkg/tfcli/types"
+)
+
+const (
+	fmtErrDestroy = "failed to destroy resource: %s"
+)
+
+// Destroy attempts to delete the resource.
+// DestroyResult.Completed is false if the operation has not yet been completed.
+func (c *client) Destroy(_ context.Context) (types.DestroyResult, error) {
+	code, tfLog, err := c.parsePipelineResult(types.OperationDestroy)
+	pipelineState, ok := cliErrors.IsPipelineInProgress(err)
```

Comment:
This was a possibility we had initially considered and after some discussions, because initiating a new pipeline in a workspace where another pipeline is already running should conceptually be considered as an error from the `tfcli` package's perspective (same as running a Terraform CLI command in a workspace where there already exists a state lock), we had decided to take this approach. And I don't see any major drawbacks here so I would prefer to stick with the existing implementation. In either implementation (in-progress pipelines being reported as errors or regular return values) the flow of action stays the same. We need to check what's in-progress, compare it with the operation being requested and take appropriate actions either by denying it or checking if a result is already available, etc.

Reply:
Alright, sounds good.